### ISSUE TITLE
Dashlist panel: tag only search failing

### ISSUE THREAD

**Issue by banburybill** [grafana/grafana#24879]
A dashboard with a dashlist panel shows a populated dashlist on Grafana 6.5.2 and previously, but the dashlist is empty after upgrading to 6.7.3 or 7.0.0. The dashlist is populated via a search using tags only.

The dashboard below illustrates the problem. The dash is tagged `testing`, and contains a dashlist that searches for dashboards in all folders tagged `testing`. The dashlist should list the containing dashboard, but is empty.

To display the expected content, change the `query` entry in the dashlist JSON from `"query": null` to `"query": ""`.

```
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": "-- Grafana --",
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "gnetId": null,
  "graphTooltip": 0,
  "id": 55,
  "links": [],
  "panels": [
    {
      "datasource": null,
      "fieldConfig": {
        "defaults": {
          "custom": {}
        },
        "overrides": []
      },
      "folderId": null,
      "gridPos": {
        "h": 4,
        "w": 24,
        "x": 0,
        "y": 0
      },
      "headings": false,
      "id": 4,
      "limit": 10,
      "query": null,
      "recent": false,
      "search": true,
      "starred": false,
      "tags": [
        "testing"
      ],
      "timeFrom": null,
      "timeShift": null,
      "title": "Dashlist",
      "type": "dashlist"
    }
  ],
  "schemaVersion": 25,
  "style": "dark",
  "tags": [
    "testing"
  ],
  "templating": {
    "list": []
  },
  "time": {
    "from": "now-6h",
    "to": "now"
  },
  "timepicker": {
    "refresh_intervals": [
      "10s",
      "30s",
      "1m",
      "5m",
      "15m",
      "30m",
      "1h",
      "2h",
      "1d"
    ]
  },
  "timezone": "",
  "title": "Test dashlist",
  "uid": "oVJdTwgMk",
  "version": 8
}
```

Server OS: Ubuntu Xenial.
Client browsers: Recent Firefox and Chrome.

**Comment by torkelo** [comment]
unable to replicate with a new panel. So might be an issue with you having query: null,  did a previous version of Grafana set that?

**Comment by banburybill** [comment]
Fair point! I've tried with the oldest Grafana I have available, 6.5.2, and it also sets "".

I'm generating the dashboard with https://github.com/weaveworks/grafanalib. This does not have a Dashlist class out of the box, so I'm using my own. Which defaults Dashlist `query` to `null`, though looking closely I now see that Grafanalib sets unspecified string items to `""`, not `null`. This worked up to and including 6.5.2, but broke when 6.7.3 was tested.

I see Grafanalib does set some items, e.g. `uid`, to `null` when not present. However, Grafana appears to  generally omit the key from JSON if not present.

So, I guess that from the Grafana point of view, this worked by accident? Should Grafanalib omit JSON keys for empty strings, or does it depend on the key - e.g. panel `title` appears to be always present? And for empty strings in e.g. panel description, is there any harm in Grafanalib emitting `"description": ""` rather than omitting it?

Thanks for the response, and many, many thanks for Grafana.

**Comment by zuchka** [comment]
@banburybill I just imported your JSON into our online sandbox, play.grafana.org, and it seems to be working as expected. The dashboard list panel identified itself via a tag search.

I am going to close this issue for now, but if you are still experiencing this, please do let us know and we can work together to drill down on the problem 👍
